checker/tests/nullness/TryWithResourcesAnno.java

```diff
@@ -0,0 +1,25 @@
+// Test case for https://tinyurl.com/cfissue/3305 .
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class TryWithResourcesAnno {
+    public static void f() {
+        try (@Nullable AutoCloseable obj = null) {
+        } catch (Exception e) {
+        }
+    }
+
+    public static void g() {
+        try (@Nullable AutoCloseable obj1 = null;
+                AutoCloseable obj2 = null) {
+        } catch (Exception e) {
+        }
+    }
+
+    public static void h() {
+        try (AutoCloseable obj1 = null;
```

Comment:
Do you want to add a test with an expected error, e.g. for `try (@Nullable final AutoCloseable obj = null) {`?

Reply:
I now see that the implementation wouldn't find such an issue.

I'm wondering a bit why the same problem doesn't exist for implicitly public methods in interfaces...

```java
interface Foo {
    @Nullable public Object get();
}
```

correctly gives a warning, but when I remove the explicit `public` the warning goes away.
Maybe the `public` isn't added to the modifiers in that situation?

Anyways, the shortcoming is documented in the method below, so we don't need a todo test.